operations/provision-consul/k8s/README.md

```diff
@@ -0,0 +1,476 @@
+    Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.6-gke.2", GitCommit:"384b4eaa132ca9a295fcb3e5dfc74062b257e7df", GitTreeState:"clean", BuildDate:"2018-08-15T00:10:14Z", GoVersion:"go1.9.3b4", Compiler:"gc", Platform:"linux/amd64"}
+
+
+### Validating Helm's Tiller on the Kubernetes cluster
+
+This guide also assumes that Helm's [Tiller](https://docs.helm.sh/glossary/#tiller) is installed and functioning properly on the k8s cluster. The [Helm website](https://helm.sh)
+has documentation on how to install and configure Helm and Helm Tiller.
+
+To validate a working Tiller (assumes the default of Tiller running in the Kubernets 'kube-system' namespace):
+
+    kubectl get pods --namespace=kube-system | grep tiller
+
+    tiller-deploy-895d57dd9-q82bp                                   1/1       Running   0          14d
+
+    helm version
+
+    Client: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
+    Server: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
+
+Your versions my differ slightly but so long as you have Running Tiller and the Client and Server versions are
+reporing reasonable values you are probably in good shape.
+
+
+### Download and install the official HashiCorp Consul Helm Chart
+
+There are a number of ways to install a Helm Chart but the process changes slightly from version to version of Helm
+and the workflow is different during Helm Chart development versus production usage. To keep things simple,
+we'll simply clone the Consul chart repo locally and drive the process from there. This is more similar to
+a development workflow but it lets us understand the workings of the Consul Helm Chart without diving too deep into
+Helm Chat workflow.
+
+    # FIXME: change this once the repo is public available
+    # git clone https://github.com/hashicorp/consul-helm ./consul-helm
+    git clone --progress git@github.com:hashicorp/consul-helm ./consul
+
+
+### Deploy (default config) Consul on Kubernetes
+
+We will use the downloaded Consul Helm Chart to deploy a Consul cluster. Note specification of
+the 'ui.service.type' parameter to the Consul Helm Chart. Specifying 'ui.service.type' allows the Operator
+to expose the Consul UI service for use outside of the Kubernetes cluster and thus makes the Consul UI
+available from a web browser.
+
+****Note****: The correct 'ui.service.type' will be highly dependent on how you your Kubernetes networking is configured.
+'LoadBalancer' works fine for default Google Container Engine-deployed Kubernetes through the default integration
+with Google Cloud's external load balancer. For other environments, 'Ingress' or 'NodePort' might make more sense.
```

Comment:
"Ingress" won't work since thats a 1st class resource type itself. The Helm chart should probably support that though it doesn't yet.